api4/user.go

```diff
@@ -111,14 +111,23 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// No permission check required
-
 	user, err := c.App.GetUser(c.Params.UserId)
 	if err != nil {
 		c.Err = err
 		return
 	}
 
+	canSee, err := c.App.UserCanSeeOtherUser(c.App.Session.UserId, c.Params.UserId)
+	if err != nil {
```

Comment:
This applies to many of these functions, but we need to check `UserCanSeeOtherUser` before calling `GetUser` as otherwise it's possible to determine whether a user exists even if you don't have permissions to view that user based on the error returned.

Reply:
Good point

Reply:
I think for now we can live it as is now, because anyway it is an improvement over the previous situation (no restriction at all). But yes, probably we should have some kind of policy here like return always the same error for both cases.

Reply:
Please can you move the `UserCanSeeOtherUser` call to *before* the `GetUser` call?

Reply:
Sure

Reply:
But there are other cases that Is more complex, when you try to get the user by username or by email, where I have to try to get the user first (not return the error anyway, and check if I can see the user), and then return the error depending on it.

Reply:
The issue is that if we ignore this, we'll get a bug bounty security ticket against it, so I think we have to get it right now, even if it's a bit complicated.

Reply:
It is already done